renderer/src/hooks/useCurrentTransactionStatus.tsx

```diff
@@ -0,0 +1,34 @@
+import { useEffect, useRef, useState } from 'react'
+import { FILTransactionProcessing } from '../../../shared/typings'
+
+const useCurrentTransactionStatus = (transaction?: FILTransactionProcessing) => {
+  const [status, setStatus] = useState<'none' | 'processing' | 'complete'>('none')
+  const [currentTransaction, setCurrentTransaction] = useState<FILTransactionProcessing>()
+  const timeout = useRef<ReturnType<typeof setTimeout>>()
+
+  useEffect(() => {
+    if (transaction) {
+      setStatus('processing')
+      setCurrentTransaction(transaction)
+    } else if (!transaction && status === 'processing') {
+      setStatus('complete')
+
+      timeout.current = setTimeout(() => {
+        setStatus('none')
+        setCurrentTransaction(undefined)
+      }, 1000)
+    }
+
+    return () => {
+      clearTimeout(timeout.current)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
could we remove use of this exemption by converting `status` into a ref? I think disabling this rule is dangerous, because for example in a next PR we could add another variable that _should_ be listed, and eslint wouldn't tell us about it.

Reply:
This sounds great to me, it's also suggested on SO here: https://stackoverflow.com/questions/63224151/how-can-i-access-state-in-an-useeffect-without-re-firing-the-useeffect

It would be great if we could go through all cases where exhaustive deps is disabled, and re-enable it through this method